lms/static/scripts/frontend_apps/components/BookList.js

```diff
@@ -0,0 +1,62 @@
+import { createElement } from 'preact';
+
+import Table from './Table';
+
+/**
+ * @typedef {import('../api-types').Book} Book
+ */
+
+/**
+ * @typedef BookListProps
+ * @prop {Book[]} books
+ * @prop {boolean} isLoading
+ * @prop {Book|null} selectedBook
+ * @prop {(b: Book) => any} onSelectBook
+ * @prop {(b: Book) => any} onUseBook
+ */
+
+/**
+ * @param {BookListProps} props
+ */
+export default function BookList({
+  books,
+  isLoading = false,
+  selectedBook,
+  onSelectBook,
+  onUseBook,
+}) {
+  const columns = [
+    {
+      label: 'Title',
+    },
+  ];
+
+  // TODO - Add a nicer loading state.
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div className="BookList">
+      <Table
+        accessibleLabel="Book list"
+        columns={columns}
+        items={books}
+        onSelectItem={onSelectBook}
+        onUseItem={onUseBook}
+        renderItem={book => <td>{book.title}</td>}
+        selectedItem={selectedBook}
+      />
+      {selectedBook && (
+        <div className="BookList__cover-container">
+          <img
```

Comment:
Its a little distracting when you go from nothing selected, to then selecting a book from the list, then the list's width shrinks to make room for the image and potentially the heigh increases if there are not too many books in the list because the image may be taller. (as is the case with only 2 books)

The other thing is the image is centered vertically, but it might look better if it was fixed to the top. I mocked up some extra books to see how it would look.

![Screen Shot 2021-05-27 at 3 15 46 PM](https://user-images.githubusercontent.com/3939074/119904048-8ce94300-befe-11eb-9faa-ba00dcd49235.png)


The other thing I noticed was that the overlfow:auto was not working. I looked at the css and I couldn't figure out why so that may be an issue with my mock data or some other css rule.

Reply:
I've revised the UI here to fix the overflow issue. The book cover now has a fixed height of 180px and the width adapts depending on the book's aspect ratio, with a minimum width for the cover container of 180px. This means that for the majority of books that are taller than they are wide, the size of the book cover container will be fixed.

Reply:
I've revised this so that the book cover container is always present, but is empty when no book is selected.